argus/client/sct/client.py

```diff
@@ -0,0 +1,156 @@
+from uuid import UUID
+import requests
+from argus.backend.util.enums import ResourceState, TestStatus
+from argus.client.base import ArgusClient
+
+
+class ArgusSCTClient(ArgusClient):
+    test_type = "scylla-cluster-tests"
+    schema_version: str | None = "v8"
+
+    class Routes(ArgusClient.Routes):
+        SUBMIT_PACKAGES = "/sct/$id/packages/submit"
+        SUBMIT_SCREENSHOTS = "/sct/$id/screenshots/submit"
+        CREATE_RESOURCE = "/sct/$id/resource/create"
+        TERMINATE_RESOURCE = "/sct/$id/resource/$name/terminate"
+        UPDATE_SHARDS_FOR_RESOURCE = "/sct/$id/resource/$name/shards"
+        SUBMIT_NEMESIS = "/sct/$id/nemesis/submit"
+        FINALIZE_NEMESIS = "/sct/$id/nemesis/finalize"
+        SUBMIT_EVENTS = "/sct/$id/events/submit"
+
+    def __init__(self, run_id: UUID, auth_token: str, base_url: str, api_version="v1") -> None:
+        super().__init__(auth_token, base_url, api_version)
+        self.run_id = run_id
+
+    def submit_sct_run(self, job_name: str, job_url: str, started_by: str, commit_id: str,
+                       runner_public_ip: str, runner_private_ip: str, sct_config: str) -> requests.Response:
+        return super().submit_run(run_type=self.test_type, run_body={
+            "run_id": str(self.run_id),
+            "job_name": job_name,
+            "job_url": job_url,
+            "started_by": started_by,
+            "commit_id": commit_id,
+            "runner_public_ip": runner_public_ip,
+            "runner_private_ip": runner_private_ip,
+            "sct_config": sct_config,
+        })
+
+    def set_sct_run_status(self, new_status: TestStatus) -> requests.Response:
+        return super().set_status(run_type=self.test_type, run_id=self.run_id, new_status=new_status)
+
+    def update_scylla_version(self, version: str) -> requests.Response:
+        return super().update_product_version(run_type=self.test_type, run_id=self.run_id, product_version=version)
+
+    def submit_sct_logs(self, logs: list[dict[str, str]]) -> requests.Response:
+        return super().submit_logs(run_type=self.test_type, run_id=self.run_id, logs=logs)
+
+    def finalize_sct_run(self) -> requests.Response:
+        return super().finalize_run(run_type=self.test_type, run_id=self.run_id)
+
+    def submit_packages(self, packages: list[dict]):
```

Comment:
`packages` - list of dict doesn't say much about structure. I propose creating dataclass Package (or more precise name) so when using api, user can easily get what is needed.

Reply:
missing return type hint

Reply:
> `packages` - list of dict doesn't say much about structure. I propose creating dataclass Package (or more precise name) so when using api, user can easily get what is needed.

I am actually thinking right now what would be better - introducing dataclasses or type hinted dicts with preset keys. Type Hinting could solve not understanding which keys you need to provide while keeping the serialization into json simple. Dataclass will most likely need an introduction of an (admittely simple) custom json encoder.

Reply:
Done, decided to settle on intermediary dataclasses